pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -65,17 +68,27 @@ func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (
 
 	context.GetLogger(ctx).Debugf("could not stat layer link %q in repository %q: %v", dgst.String(), repo.Named().Name(), err)
 
-	// verify the blob is stored locally
-	desc, err = dockerRegistry.BlobStatter().Stat(ctx, dgst)
-	if err != nil {
-		return desc, err
+	// only non-empty layers is wise to check for existence in the image stream.
+	// schema v2 has no empty layers.
+	if dgst.String() != digestSHA256GzippedEmptyTar.String() {
+		// ensure it's referenced inside of corresponding image stream
+		if !imageStreamHasBlob(repo, dgst) {
+			return distribution.Descriptor{}, distribution.ErrBlobUnknown
```

Comment:
Consider:

1. empty image stream `ns/foo`
2. remote manifest with layers `{A}` tagged as `docker.io/foo:v1`
3. remote manifest with layers `{B}` tagged as `docker.io/foo:v2`

and following scenario:

```sh
oc import-image --from=docker.io/foo:v1 foo:v1 --confirm
docker pull docker.io/foo:v2
docker tag docker.io/foo:v2 "${integrated_registry_url}/ns/foo:v2"
docker push "${integrated_registry_url}/ns/foo:v2"
```

This will fail because `HEAD` on `B` in `ns/foo` will succeed thanks to pullthrough. However, manifest verification will fail because `B` is not tagged in the image stream and it's not an empty layer.